src/AstRunner/AstMap/AstDependency.php

```diff
@@ -35,73 +52,8 @@ public function getType(): string
         return $this->type;
     }
 
-    public static function useStmt(ClassLikeName $classLikeName, FileOccurrence $fileOccurrence): self
-    {
-        return new self($classLikeName, $fileOccurrence, 'use');
-    }
-
-    public static function returnType(ClassLikeName $classLikeName, FileOccurrence $fileOccurrence): self
-    {
-        return new self($classLikeName, $fileOccurrence, 'returntype');
-    }
-
-    public static function parameter(ClassLikeName $classLikeName, FileOccurrence $fileOccurrence): self
-    {
-        return new self($classLikeName, $fileOccurrence, 'parameter');
-    }
-
-    public static function newStmt(ClassLikeName $classLikeName, FileOccurrence $fileOccurrence): self
-    {
-        return new self($classLikeName, $fileOccurrence, 'new');
-    }
-
-    public static function staticProperty(ClassLikeName $classLikeName, FileOccurrence $fileOccurrence): self
-    {
-        return new self($classLikeName, $fileOccurrence, 'static_property');
-    }
-
-    public static function staticMethod(ClassLikeName $classLikeName, FileOccurrence $fileOccurrence): self
-    {
-        return new self($classLikeName, $fileOccurrence, 'static_method');
-    }
-
-    public static function instanceofExpr(ClassLikeName $classLikeName, FileOccurrence $fileOccurrence): self
-    {
-        return new self($classLikeName, $fileOccurrence, 'instanceof');
-    }
-
-    public static function catchStmt(ClassLikeName $classLikeName, FileOccurrence $fileOccurrence): self
-    {
-        return new self($classLikeName, $fileOccurrence, 'catch');
-    }
-
-    public static function variable(ClassLikeName $classLikeName, FileOccurrence $fileOccurrence): self
-    {
-        return new self($classLikeName, $fileOccurrence, 'variable');
-    }
-
-    public static function throwStmt(ClassLikeName $classLikeName, FileOccurrence $fileOccurrence): self
-    {
-        return new self($classLikeName, $fileOccurrence, 'throw');
-    }
-
-    public static function constFetch(ClassLikeName $classLikeName, FileOccurrence $fileOccurrence): self
-    {
-        return new self($classLikeName, $fileOccurrence, 'const');
-    }
-
-    public static function anonymousClassExtends(ClassLikeName $classLikeName, FileOccurrence $fileOccurrence): self
-    {
-        return new self($classLikeName, $fileOccurrence, 'anonymous_class_extends');
-    }
-
-    public static function anonymousClassImplements(ClassLikeName $classLikeName, FileOccurrence $fileOccurrence): self
-    {
-        return new self($classLikeName, $fileOccurrence, 'anonymous_class_implements');
-    }
-
-    public static function attribute(ClassLikeName $classLikeName, FileOccurrence $fileOccurrence): self
+    public static function fromType(TokenName $tokenName, FileOccurrence $fileOccurrence, string $type): self
     {
-        return new self($classLikeName, $fileOccurrence, 'attribute');
+        return new self($tokenName, $fileOccurrence, $type);
```

Comment:
Well actually, I do require them now, so that is why I changed it. I use them in the emitters to differentiate the type of `AstDependency` and whether to convert to `Dependency`

Reply:
I would like to convert them to `enum` in the future but that would either require PHP 8.1 or some enum library.